### ISSUE TITLE
Async can resolve / reject more than once.

### ISSUE THREAD

**Issue by JohanCodinha** [evilsoft/crocks#293]
Given this bit of code:
```js
const { Async } = require('crocks')
const log = tag => x => console.log(`${tag}: ${x}`)
const crocksAsync = Async((rej, res) => {
    res('first')
    res('second')
    rej('third')
    res('fourth')
})
crocksAsync.fork(log('rej'), log('res'))
```

Expected console output
```js
"res: first"
```
Actually console output
```js
"res: first"
"res: second"
"rej: third"
"res: fourth"
```

**Comment by evilsoft** [comment]
Start on this.

**Comment by evilsoft** [comment]
Boy this is a tricky will, will try to have it out in the next release, but it may take (2).

**Comment by evilsoft** [comment]
Submitted [this PR](https://github.com/evilsoft/crocks/pull/309) to address this bug!
Looks like if all is good with the PR we will be getting this out on the next release.

@JohanCodinha Thanks again for the 🐞. Much obliged!!

**Comment by evilsoft** [comment]
@JohanCodinha This fix is now published and available in `v0.10.2`.
Would like to mention you on twitter for the bug report when announcing this release. If that is cool, please let me know you twitter handle.

**Comment by evilsoft** [comment]
Shipped with `0.10.2` closing.

**Comment by JohanCodinha** [comment]
> @JohanCodinha This fix is now published and available in `v0.10.2`.
> Would like to mention you on twitter for the bug report when announcing this release. If that is cool, please let me know you twitter handle.

Thanks for mate ! @Panelvw
